Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 250)
+++ CMakeLists.txt	(working copy)
@@ -51,4 +51,4 @@

 add_subdirectory(quazip)

-install(FILES FindQuaZip.cmake DESTINATION ${CMAKE_ROOT}/Modules)
+#install(FILES FindQuaZip.cmake DESTINATION ${CMAKE_ROOT}/Modules)
Index: quazip/CMakeLists.txt
===================================================================
--- quazip/CMakeLists.txt	(revision 250)
+++ quazip/CMakeLists.txt	(working copy)
@@ -14,10 +14,14 @@
 qt_wrap_cpp(MOC_SRCS ${PUBLIC_HEADERS})
 set(SRCS ${SRCS} ${MOC_SRCS})

-add_library(quazip SHARED ${SRCS})
-set_target_properties(quazip PROPERTIES VERSION 1.0.0 SOVERSION 1)
+add_library(quazip STATIC ${SRCS})
+#set_target_properties(quazip PROPERTIES VERSION 1.0.0 SOVERSION 1)
 # Link against ZLIB_LIBRARIES if needed (on Windows this variable is empty)
-target_link_libraries(quazip ${QT_QTMAIN_LIBRARY} ${QT_QTCORE_LIBRARY} ${ZLIB_LIBRARIES})
+target_link_libraries(quazip ${ZLIB_LIBRARIES})
+qt5_use_modules(quazip Core)
+if(WIN32)
+    add_definitions(-DZ_PREFIX)
+endif()

 install(FILES ${PUBLIC_HEADERS} DESTINATION include/quazip)
 install(TARGETS quazip LIBRARY DESTINATION ${LIB_DESTINATION} ARCHIVE DESTINATION ${LIB_DESTINATION} RUNTIME DESTINATION ${LIB_DESTINATION})