diff --git a/src/init.c b/src/init.c
index 06dbb3f2..a7c6da86 100644
--- a/src/init.c
+++ b/src/init.c
@@ -449,6 +449,12 @@ GLFWAPI int glfwInit(void)
     _glfw.initialized = GLFW_TRUE;
 
     glfwDefaultWindowHints();
+
+    fprintf(stderr, "!!! Patched GLFW from https://github.com/Admicos/minecraft-wayland\n"
+         "!!! If any issues with the window, or some issues with rendering, occur, "
+         "first try with the built-in GLFW, and if that solves the issue, report there first.\n"
+         "!!! Use outside Minecraft is untested, and things might break.\n");
+
     return GLFW_TRUE;
 }
 
